apps/ios/Shared/ContentView.swift

```diff
@@ -48,28 +48,19 @@ struct ContentView: View {
                     }
                 }
             } else if prefPerformLA && laFailed {
-                retryAuthView()
-            }
-        }
-        .onChange(of: doAuthenticate) { doAuth in
-            if doAuth, authenticationExpired() {
-                runAuthenticate()
+                Button(action: runAuthenticate) { Label("Retry", systemImage: "arrow.counterclockwise") }
+            } else {
+                Button(action: runAuthenticate) { Image(systemName: "lock.open") }
             }
         }
+        .onAppear(perform: runAuthenticate)
+        .onChange(of: doAuthenticate) { _ in runAuthenticate() }
         .alert(isPresented: $alertManager.presentAlert) { alertManager.alertView! }
     }
 
-    private func retryAuthView() -> some View {
-        Button {
-            laFailed = false
-            runAuthenticate()
-        } label: { Label("Retry", systemImage: "arrow.counterclockwise") }
-    }
-
     private func runAuthenticate() {
-        if !prefPerformLA {
```

Comment:
Why this had to be changed? If `prefPerformLA` is false we don't need other checks to set `userAuthorized = true`.. Then its else branch it could check for `if doAuthenticate && authenticationExpired()` separately